File: java/0014_longestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String res = strs[0];
        for(int i=1;i<strs.length;i++)
            res = getCommon(res,strs[i]);
        return res;
    }
    public String getCommon(String s1,String s2)
    {
        String str = "";
        for(int i=0;i<Math.min(s1.length(),s2.length())&&s1.charAt(i)==s2.charAt(i);i++)
        {
            str+=s1.charAt(i);
        }
        return str;
    }
}

File: cpp/0014_longestCommonPrefix.cpp
// 直接横向反复取交集
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res = strs[0];
        for(auto i=1;i<strs.size();i++)
            res = getCommon(strs[i],res);
        return res;
    }
    string getCommon(string s1,string s2)
    {
        string str="";
        for(int i=0;i<min(s1.length(),s2.length()) && s1[i]==s2[i];i++)
            str+=s1[i];
        return str;
    }
};
